### ISSUE TITLE
Stratholme Gauntlet Run

### ISSUE THREAD

**Issue by scotty0100** [scriptdev2/scriptdev2-classic#43]
How to replicate the bug: do the gauntlet run
What happens: once you get to the Baron and still have enough time, you engage him but he still kills the npc anyway
What should happen: once you engage him, the npc should live therefore completing the gauntlet and associated quest
Quest: http://www.wowwiki.com/Quest:Dead_Man%27s_Plea

Extra Information:
"Baron Rivendare
You do not have to kill the Baron in 45 minutes, you only have to engage him in 45 minutes. However, if you wipe after engaging him, you will fail the quest. From this point, it is exactly the same as on a normal Baron run. After you have taken him down, you can talk to Ysida and finish the quest."

We did not wipe, but he still killed the Npc.

**Comment by xfurry** [comment]
So basically you are saying that the Baron Gauntlet Run timer should be put on hold while the encounter is in progress?

**Comment by scotty0100** [comment]
Yes - but also the NPC is never actually ever spawned. The cage is empty.

The dialogue goes through as if she is being killed anyway. Doesn't matter if you complete it within 20 minutes, or 1 hour it always does the same thing.

The script seems to keep running even if you've killed the Baron as well, so it carries on saying he kills her after he's been killed

**Comment by xfurry** [comment]
Yeah, I can see that the handling of this event is pretty poor.
Will fix it in a future commit on master, then will be backported.

**Comment by Phatcat** [comment]
[This recent commit](https://github.com/cmangos/mangos-classic/commit/cf63af912c3e7d2891903854d1ef6eca64c835a1) should hopefully close this issue.

**Comment by cala** [comment]
Yes, the Baron run in Stratholme is now correctly handled, including what is reported here.
